http/auth/migration_script.up.sql

```diff
@@ -0,0 +1,42 @@
+CREATE TABLE IF NOT EXISTS auth_roles
+(
+    id         BIGSERIAL PRIMARY KEY,
+    name       TEXT NOT NULL,
+    created_at TIMESTAMP,
+    updated_at TIMESTAMP,
+    deleted_at TIMESTAMP
+);
+
+
+CREATE TABLE IF NOT EXISTS auth_users
+(
+    id         BIGSERIAL PRIMARY KEY,
+    email      TEXT NOT NULL,
+    created_at TIMESTAMP,
+    updated_at TIMESTAMP,
+    deleted_at TIMESTAMP
+);
+
+CREATE TABLE IF NOT EXISTS auth_user_roles
+(
+    id         BIGSERIAL PRIMARY KEY,
+    role_id    BIGINT REFERENCES auth_roles (id) NOT NULL,
+    user_id    BIGINT REFERENCES auth_users (id) NOT NULL,
+    created_at TIMESTAMP,
+    updated_at TIMESTAMP,
+    deleted_at TIMESTAMP
+);
+
+CREATE TABLE IF NOT EXISTS auth_role_permissions
+(
+    id         BIGSERIAL PRIMARY KEY,
+    role_id    BIGINT REFERENCES auth_roles (id) NOT NULL,
+    resource   TEXT                              NOT NULL,
+    method     TEXT                              NOT NULL,
+    created_at TIMESTAMP,
+    updated_at TIMESTAMP,
+    deleted_at TIMESTAMP
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_auth_roles_name ON auth_roles (name);
+CREATE UNIQUE INDEX IF NOT EXISTS idx_auth_users_email ON auth_users (email);
```

Comment:
add the following unique index?
- auth_user_roles with role_id and user_id
- auth_role_permissions with role_id, resource and method